### ISSUE TITLE
heap over flow

### ISSUE THREAD

**Issue by bird8693** [crupp2/png2gif#1]
# enviroment
ubuntu 18.04

# requirement
compile with asan

# command
./png2gif mm.gif poc

# poc
https://drive.google.com/file/d/1uJtlpuygYg0kGx4_CexwBc6fI3dhLA_5/view?usp=sharing

# output
```
==33070==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x602000000011 at pc 0x7f4e42aec733 bp 0x7ffe902bd5b0 sp 0x7ffe902bcd58
READ of size 32 at 0x602000000011 thread T0
    #0 0x7f4e42aec732  (/usr/lib/x86_64-linux-gnu/libasan.so.4+0x79732)
    #1 0x55bf6c1ca9dd in writeGIFLCT /home/fuzz/png2gif/gifWriter.c:486
    #2 0x55bf6c1cb441 in writeGIFFrame /home/fuzz/png2gif/gifWriter.c:161
    #3 0x55bf6c1b7a9a in main /home/fuzz/png2gif/png2gif.c:177
    #4 0x7f4e41ee5bf6 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21bf6)
    #5 0x55bf6c1b8909 in _start (/home/micro/all/found/png2gif/png2gif_asan+0x10909)

0x602000000011 is located 0 bytes to the right of 1-byte region [0x602000000010,0x602000000011)
allocated by thread T0 here:
    #0 0x7f4e42b51d28 in __interceptor_calloc (/usr/lib/x86_64-linux-gnu/libasan.so.4+0xded28)
    #1 0x55bf6c1c8aa8 in writeGIFLCT /home/fuzz/png2gif/gifWriter.c:462
    #2 0x55bf6c1cb441 in writeGIFFrame /home/fuzz/png2gif/gifWriter.c:161
    #3 0x55bf6c223899  (/home/micro/all/found/png2gif/png2gif_asan+0x7b899)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/usr/lib/x86_64-linux-gnu/libasan.so.4+0x79732) 
Shadow bytes around the buggy address:
  0x0c047fff7fb0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c047fff7fc0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c047fff7fd0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c047fff7fe0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c047fff7ff0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0c047fff8000: fa fa[01]fa fa fa 01 fa fa fa fa fa fa fa fa fa
  0x0c047fff8010: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8020: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8030: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8040: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c047fff8050: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==33070==ABORTING

```

**Comment by afterdelight** [comment]
ngentot

**Comment by crupp2** [comment]
This issue was caused by allowing images with zero width/height. Added input checks to prevent processing such images. Fixed in 323c5de.
Thank you for reporting this.
